### PR TITLE
Add simple attribute completions for all tags and many attribute values on the HTMX reference

### PR DESCRIPTION
**Pull request by itepastra**
Adding attribute completions for the hx-trigger and hx-boost tags.

I used the documentation on htmx.org where it was applicable.
I did spot a typo in the documentation which I fixed.
For the attributes that did not have a relevant entry in the documentation I wrote some simple explanation myself.

### CODE REVIEW

File: lsp/src/htmx/attributes/hx-ext.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+reference an extension
```
Comment (cristianoliveira) [inline lsp/src/htmx/attributes/hx-ext.md:1]: I noticed https://github.com/microsoft/vscode-html-languageservice links the source of each completion with the source material like:
Screenshot 2023-08-28 at 09 33 12

Shall we do the same? WDYT?

Comment (itepastra) [reply lsp/src/htmx/attributes/hx-ext.md:1]: I think this might be a good idea, will add those in a bit.

**Comment by cristianoliveira** [comment]
I read through the docs and they look good. The only small fix I'd suggest is to align the style of code blocks, some have a [small empty + tab like this one](https://github.com/ThePrimeagen/htmx-lsp/pull/9/files#diff-a44a1be74ea03f56348879c1799fe793a46c7996fa5f5209c5db20eaf50de16aR69) and others [don't](https://github.com/ThePrimeagen/htmx-lsp/pull/9/files#diff-a44a1be74ea03f56348879c1799fe793a46c7996fa5f5209c5db20eaf50de16aR58)
Screenshot 2023-08-28 at 10 33 04

I like the spaced code examples, but this is very much a nitpick 😄

**Comment by itepastra** [comment]
I mostly copied from the documentation for this pr, will have another look to see if I can get it to be more consistent.

**Comment by itepastra** [comment]
also I didn't know there was an "indent selection" in nvim until I was nearly done. So I got annoyed with doing it manually lol.

**Comment by itepastra** [comment]
> I read through the docs and they look good. The only small fix I'd suggest is to align the style of code blocks, some have a [small empty + tab like this one](https://github.com/ThePrimeagen/htmx-lsp/pull/9/files#diff-a44a1be74ea03f56348879c1799fe793a46c7996fa5f5209c5db20eaf50de16aR69) and others [don't](https://github.com/ThePrimeagen/htmx-lsp/pull/9/files#diff-a44a1be74ea03f56348879c1799fe793a46c7996fa5f5209c5db20eaf50de16aR58) Screenshot 2023-08-28 at 10 33 04
> 
> I like the spaced code examples, but this is very much a nitpick 😄

I've improved the indenting of the code examples where I could quickly see errors in https://github.com/ThePrimeagen/htmx-lsp/pull/9/commits/835d4f71787ec79b64cbd740fd21d8916893e128.

**Comment by itepastra** [comment]
@cristianoliveira, I'm wondering whether I should make all lists just indented, or with `*`'s in front. What would you suggest?

**Comment by itepastra** [comment]
It now includes ever attribute on https://htmx.org/reference/, when attributes had a listed value on the site like https://htmx.org/attributes/hx-swap-oob/ those values are also included. All documentation is either copied or paraphrased from the htmx website.

Deprecated items are intentionally not included.

File: lsp/src/htmx/attributes/hx-history-elt.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+Here is an example:
+
+<html>
+<body>
+<div id="content" hx-history-elt>
+ ...
+</div>
+</body>
+</html>
```
Comment (cristianoliveira) [inline lsp/src/htmx/attributes/hx-history-elt.md:11]: Missing format :)
```suggestion
<html>
  <body>
    <div id="content" hx-history-elt>
     ...
    </div>
  </body>
</html>
```

File: lsp/src/htmx/attributes/hx-history.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+Here is an example:
+
+<html>
+<body>
+<div hx-history="false">
+ ...
+</div>
+</body>
+</html>
```
Comment (cristianoliveira) [inline lsp/src/htmx/attributes/hx-history.md:13]:
```suggestion
<html>
  <body>
    <div hx-history="false">
     ...
    </div>
  </body>
</html>
```

**Comment by cristianoliveira** [comment]
LGTM 

Eventually, I believe formatting stuff could be automated in CI

**Comment by ThePrimeagen** [comment]
ok, this is great

looking now.  i also want to develop a nice CI next.

thanks fro getting through the "exploration phase"

**Comment by ThePrimeagen** [comment]
this is the "impossible" review.  but the mod.rs file looks correct.

i am positive there will be a bug report but this is an AMAZING effort.  thank you so much

**Pull request merged**